frame/staking/src/tests.rs

```diff
@@ -2894,6 +2893,11 @@ mod offchain_phragmen {
 		)
 	}
 
+	#[test]
+	fn offchain_phragmen_with_less_than_minimum_candidates() {
+		todo!()
```

Comment:
I think current offchain worker code does not check for this. we need `x` validators and we have `c` candidates where `c < x`, then all the functions in `sp-npos-elections` will return `c` without any error (as defined in their API). And then the offchain worker will try and submit this, which can cause the solution to be invalid. 

While it is very unlikely to happen, it could happen and we should prevent. This can be moved to another issue as well.